### PR TITLE
Update password hash documentation on basic authentication

### PR DESCRIPTION
**Pull request by BitPatty**
https://issues.apache.org/jira/browse/SOLR-XXXXX

# Description

The algorithm for the password hash in the basic auth plugin docs is incorrect:

```diff
- sha256(password+salt)
+ sha256(sha256(salt+password))
```

See:

https://github.com/apache/solr/blob/9d6306e061374a4dea0dd14c7397994e1ab0bebe/solr/core/src/java/org/apache/solr/security/Sha256AuthenticationProvider.java#L124-L132

# Solution

Updated docs.

# Tests

Didn't run a build due to missing setup.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [x] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Review by epugh — COMMENTED** [review]
LGTM, but I dont' have a lot of experience in this area, maybe @janhoy  you want to give a once over?

File: solr/solr-ref-guide/modules/deployment-guide/pages/basic-authentication-plugin.adoc [hunk 1]
```diff
@@ -55,7 +55,7 @@ There are several options defined in this example:
 
 <1> The first block defines the authentication plugin to be used and its parameters.
 <2> The parameter `"blockUnknown":true` means that unauthenticated requests are not allowed to pass through.
-<3> A user called 'solr', with a password `'SolrRocks'` has been defined.
+<3> A user called 'solr', with a password `'SolrRocks'` in the format `base64(<password hash>) base64(<salt>)` has been defined.
```
Comment (janhoy) [inline solr/solr-ref-guide/modules/deployment-guide/pages/basic-authentication-plugin.adoc:58]: No need to repeat this detail here again.
```suggestion
<3> A user called 'solr', with a password `'SolrRocks'`, in the encoded format detailed above, has been defined.
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/basic-authentication-plugin.adoc [hunk 2]
```diff
@@ -29,7 +29,7 @@ This file and where to put it is described in detail in the section xref:authent
 If running in cloud mode, you can use the `bin/solr auth` command-line utility to enable security for a new installation, see: `bin/solr auth --help` for more details.
 
 For Basic authentication, `security.json` must have an `authentication` block which defines the class being used for authentication.
-Usernames and passwords (as a sha256(password+salt) hash) could be added when the file is created, or can be added later with the Authentication API, described below.
+Usernames and passwords (as a base64 encoded `sha256(sha256(salt+password))` hash) could be added when the file is created, or can be added later with the Authentication API, described below.
```
Comment (janhoy) [inline solr/solr-ref-guide/modules/deployment-guide/pages/basic-authentication-plugin.adoc:32]:
```suggestion
Usernames and passwords (Format: `base64(sha256(sha256(salt+password)) base64(salt)`) could be added when the file is created, or can be added later with the Authentication API, described below.
```

**Review by janhoy — COMMENTED** [review]
I have a suggestion for change. Better reveal the whole formulae first, and then refer to it in the bullet list?

**Comment by janhoy** [comment]
@epugh You may find https://github.com/clemente-biondo/clemente-biondo.github.io/blob/master/solr_pwd_encryption.js interesting, it is JS code which is perhaps easier to read to understand the encoding.

**Comment by epugh** [comment]
Thanks @BitPatty  and @janhoy  for this improvement!

**Pull request merged**